cf/formula_terms.py

```diff
@@ -0,0 +1,2057 @@
+'''Functions for computing non-parametric vertical coordinates from
+the formula defined by a coordinate reference construct.
+
+See Appendix D: Parametric Vertical Coordinates of the CF conventions.
+
+.. versionaddedd:: 3.8.0
+
+'''
+import logging
+
+from .units import Units
+
+from .functions import bounds_combination_mode
+
+from .constants import (
+    formula_terms_standard_names,
+    formula_terms_max_dimensions,
+    formula_terms_units,
+    formula_terms_computed_standard_names,
+    formula_terms_D1,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _domain_ancillary_term(f, standard_name, coordinate_conversion,
+                           term, default_to_zero, strict, bounds):
+    '''Find a domain ancillary construct cooresponding to a formula term.
+
+    .. versionadded:: 3.8.0
+
+    :Parameters:
+
+        f: `Field`
+            The parent field construct.
+
+        standard_name: `str`
+            The standard name of the parametric vertical coordinate.
+
+        coordinate_conversion: `CoordinateConversion`
+            The definition of the formula
+
+        term: `str`
+            A term of the formula.
+
+            *Parameter example:*
+              ``term='orog'``
+
+        default_to_zero: `bool`, optional
+            If False then do not assume that missing terms have a
```

Comment:
Good catch. It should have said:

```If False then do not assume that missing terms have a
            value of zero. By default a missing term is assumed to be
            zero, as described in Appendix D: Parametric Vertical
            Coordinates of the CF conventions.
    
```
like all the other occurences. A clear candidate for docstring rewriting. For that we'd have to convert all of the functions in `formula_term.py` into methods of a `FormulaTerms` class  - a trivial exercise.

Shall I do that?

Reply:
Please do, assuming it isn't too much bother right now, we could always do that as follow-on work but it might be less work anyway to do off the bat than to amend all the occurrences written out separately.